packages/@aws-cdk/aws-applicationsignals-alpha/lib/slo/slo-types.ts

```diff
@@ -0,0 +1,138 @@
+        return new RequestBasedSlo(scope, id, props);
+    }
+
+    protected createBurnRateConfigurations(): applicationsignals.CfnServiceLevelObjective.BurnRateConfigurationProperty[] | undefined {
+        return this.props.burnRateWindows?.map(minutes => ({
+            lookBackWindowMinutes: minutes,
+        }));
+    }
+
+    protected getMetricComparisonOperator(metric: PeriodBasedAppSignalsMetricProps | PeriodBasedCloudWatchMetricProps | RequestBasedAppSignalsMetricProps | RequestBasedCloudWatchMetricProps): ComparisonOperator {
+        return metric.comparisonOperator ?? this.getDefaultComparisonOperator(metric.metricType);
+    }
+
+    protected getPeriod(periodSeconds?: number): number {
+        return periodSeconds ?? 60;
+    }
+
+    protected isAppSignalsMetric(metric: PeriodBasedAppSignalsMetricProps | PeriodBasedCloudWatchMetricProps | RequestBasedAppSignalsMetricProps | RequestBasedCloudWatchMetricProps): boolean {
+        return 'metricType' in metric && 'keyAttributes' in metric;
+    }
+
+    protected getDefaultComparisonOperator(metricType?: MetricType): ComparisonOperator {
+        switch (metricType) {
+            case MetricType.LATENCY:
+                return ComparisonOperator.LESS_THAN_OR_EQUAL;
+            case MetricType.AVAILABILITY:
+                return ComparisonOperator.GREATER_THAN_OR_EQUAL;
+            default:
+                throw new Error('ComparisonOperator must be specified when metricType is not provided');
+        }
+    }
+    abstract _bind(): applicationsignals.CfnServiceLevelObjectiveProps;
+}
+
+/**
+ * Period-based slo
+ */
+export class PeriodBasedSlo extends ServiceLevelObjective {
+    constructor(scope: Construct, id: string, props: PeriodBasedSloProps) {
+        super(scope, id, props);
+    }
+
+    _bind(): applicationsignals.CfnServiceLevelObjectiveProps {
+        const metric = this.props.metric;
+        return {
+            name: this.name,
+            description: this.props.description,
+            goal: this.goal._bind(),
+            burnRateConfigurations: this.createBurnRateConfigurations(),
+            sli: {
+                sliMetric: this.isAppSignalsMetric(metric) ? {
+                    metricType: metric.metricType,
+                    keyAttributes: metric.keyAttributes,
+                    operationName: metric.operationName,
+                    periodSeconds: this.getPeriod(metric.periodSeconds),
+                    statistic: metric.statistic,
+                } : {
+                    metricDataQueries: metric.metricDataQueries,
+                    periodSeconds: this.getPeriod(metric.periodSeconds),
+                    statistic: metric.statistic,
+                },
+                comparisonOperator: this.getMetricComparisonOperator(metric),
+                metricThreshold: metric.metricThreshold,
+            },
+        };
+    }
+}
+
+/**
+ * Request-based slo
+ */
+export class RequestBasedSlo extends ServiceLevelObjective {
```

Comment:
Add a comment mentioning that this is an L2 construct representing which CFN resource.